test/WebSocketSession.hpp

```diff
@@ -56,9 +56,9 @@ class WebSocketSession final : public WebSocketHandler
         return result;
     }
 
-    WebSocketSession(const std::string& hostname, Protocol protocolType, int portNumber)
+    WebSocketSession(std::string && hostname, Protocol protocolType, int portNumber)
```

Comment:
Why not take a straight copy (`std::string hostname`)? At least that's the common pattern I'm familiar with (and used extensively in the code). Is there an advantage to taking r-value reference?

Reply:
Yeah .. this is a specific private ctor only, so moving the string is sufficient.
Also just our test tool. I would have thought longer about it, if public API.
For the latter, I am sure I have seen other occurrences .. and avoiding copy (elision if super fine though) is good.